interop-testing/src/main/java/io/grpc/testing/integration/StressTestClient.java

```diff
@@ -0,0 +1,574 @@
+
+  private static ManagedChannel createChannel(InetSocketAddress address) {
+    return NettyChannelBuilder.forAddress(address)
+        .negotiationType(NegotiationType.PLAINTEXT)
+        .build();
+  }
+
+  private static String serverAddressesToString(List<InetSocketAddress> addresses) {
+    return Joiner.on(",").join(Lists.transform(addresses,
+        new Function<InetSocketAddress, String>() {
+          @Nullable
+          @Override
+          public String apply(@Nullable InetSocketAddress input) {
+            Preconditions.checkNotNull(input);
+            return input.getHostName() + ":" + input.getPort();
+          }
+        }));
+  }
+
+  private static String validTestCasesHelpText() {
+    StringBuilder builder = new StringBuilder();
+    for (TestCases testCase : TestCases.values()) {
+      String strTestcase = testCase.name().toLowerCase();
+      builder.append("\n      ")
+          .append(strTestcase)
+          .append(": ")
+          .append(testCase.description());
+    }
+    return builder.toString();
+  }
+
+  /**
+   * A stress test worker. Every stub has its own stress test worker.
+   */
+  private class Worker implements Runnable {
+
+    // Interval at which the QPS stats of metrics service are updated.
+    private static final long METRICS_COLLECTION_INTERVAL_SECS = 5;
+
+    private final ManagedChannel channel;
+    private final List<TestCaseWeightPair> testCaseWeightPairs;
+    private final int durationSec;
+    private final String gaugeName;
+
+    Worker(ManagedChannel channel, List<TestCaseWeightPair> testCaseWeightPairs,
+        int durationSec, String gaugeName) {
+      this.channel = channel;
+      this.testCaseWeightPairs = testCaseWeightPairs;
+      this.durationSec = durationSec;
+      this.gaugeName = gaugeName;
+    }
+
+    @Override
+    public void run() {
+      // Simplify debugging if the worker crashes / never terminates.
+      Thread.currentThread().setName(gaugeName);
+
+      final Tester tester = new Tester();
+      tester.setUp();
+      final WeightedTestCaseSelector testCaseSelector =
+          new WeightedTestCaseSelector(testCaseWeightPairs);
+      final long endTime = durationSec < 0
+          ? -1
+          : System.nanoTime() + SECONDS.toNanos(durationSec);
+
+      // Set it so that the stats are published after the first loop iteration.
+      long lastMetricsCollectionTime = System.nanoTime()
+          - SECONDS.toNanos(METRICS_COLLECTION_INTERVAL_SECS);
+      long testCasesSinceLastMetricsCollection = 0;
+      while (!shutdown && (endTime == -1 || endTime > System.nanoTime())) {
+        try {
+          runTestCase(tester, testCaseSelector.nextTestCase());
+        } catch (Exception e) {
+          e.printStackTrace();
+          throw new RuntimeException(e);
+        }
+
+        testCasesSinceLastMetricsCollection++;
+
+        if (lastMetricsCollectionTime
+            + SECONDS.toNanos(METRICS_COLLECTION_INTERVAL_SECS) <= System.nanoTime()) {
```

Comment:
here too.  nanotime can be negative, so  you should have something like:

    lastMetricsCollectionTime  - System.nanoTime() < 0

to avoid overflow problems.